java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -84,6 +85,36 @@ public JavaType resolveTypeSubstitution(JavaType type, JavaType definition) {
     return typeSubstitutionSolver.applySiteSubstitution(type, definition);
   }
 
+  public JavaType resolveTypeSubstitutionWithDiamondOperator(ParametrizedTypeJavaType type, JavaType definition) {
+    if (isParametrizedType(definition)) {
+      ParametrizedTypeJavaType parametrizedDefinition = (ParametrizedTypeJavaType) definition;
+      JavaType newType = type;
+      if (type.rawType == parametrizedDefinition.rawType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)